azure-pipelines.yml

```diff
@@ -1,342 +1,416 @@
-      value: 'azure-control-stg-rg'
-    ${{ if notIn( parameters['env'], 'aat', 'perftest', 'preview') }}:
-      value: 'azure-control-${{ parameters.env }}-rg'
-  - name: aksKeyVault
-    ${{ if eq( parameters['env'], 'perftest') }}:
-      value: "dcdcftappstestkv"
-    ${{ if eq( parameters['env'], 'aat' ) }}:
-      value: "dcdcftappsstgkv"
-    ${{ if eq( parameters['env'], 'ptlsbox' ) }}:
-      value: "dtscftptlsbox"
-    ${{ if eq( parameters['env'], 'ptl' ) }}:
-      value: "dtscftptl"
-    ${{ if eq( parameters['env'], 'preview' ) }}:
-      value: "dcdcftappsdevkv"
-    ${{ if notIn( parameters['env'], 'aat', 'perftest', 'ptlsbox', 'ptl', 'preview') }}:
-      value: dcdcftapps${{ parameters.env }}kv
-  - name: terraformSubscriptionID
-    ${{ if eq( parameters['env'], 'sbox') }}:
-      value: "b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb"
-    ${{ if eq( parameters['env'], 'ptlsbox') }}:
-      value: "1497c3d7-ab6d-4bb7-8a10-b51d03189ee3"
-    ${{ if eq( parameters['env'], 'perftest') }}:
-      value: "8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c"
-    ${{ if eq( parameters['env'], 'ithc') }}:
-      value: "62864d44-5da9-4ae9-89e7-0cf33942fa09"
-    ${{ if eq( parameters['env'], 'demo') }}:
-      value: "d025fece-ce99-4df2-b7a9-b649d3ff2060"      
-    ${{ if eq( parameters['env'], 'ptl') }}:
-      value: "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-    ${{ if eq( parameters['env'], 'preview') }}:
-      value: "8b6ea922-0862-443e-af15-6056e1c9b9a4"
-    ${{ if eq( parameters['env'], 'aat') }}:
-      value: "96c274ce-846d-4e48-89a7-d528432298a7"
-    ${{ if eq( parameters['env'], 'prod') }}:
-      value: "8cbc6f36-7c56-4963-9d36-739db5d00b27"   
-
 
 stages:
-
-  - stage: PreChecks
-    displayName: 'PreChecks'
+  - stage: PreCheck
+    displayName: "PreChecks"
     jobs:
       - job: PreChecks
         steps:
           - template: steps/terraform-precheck.yaml@cnp-azuredevops-libraries
             parameters:
               keyvaultName: infra-vault-nonprod
-              keyvaultSecret: 'azure-devops-sp-token'
+              keyvaultSecret: "azure-devops-sp-token"
               serviceConnection: azurerm-sandbox
               projectName: ${{ variables.project }}
               cluster: ${{ parameters.cluster }}
-              environment: ${{ parameters.env }}
+              environment: sbox
               runManualStart: true
+  - ${{ each component in parameters.environment_components }}:
+      - stage: "CheckingClusters_${{ component.env }}"
+        displayName: "Checking Clusters for ${{ component.env }}"
+        dependsOn:
+          - ${{ if ne(component.env, 'sbox') }}:
+              - "Network_${{ component.dependsOn }}"
+              - "Managed_Identity_${{ component.dependsOn }}"
+              - "Aks_${{ component.dependsOn }}"
+              - "PipelineTests_${{ component.dependsOn }}"
+              - "BootStrapClusters_${{ component.dependsOn }}"
+          - ${{ else }}:
+              - "Precheck"
```

Comment:
> Could you explain this a little? Each stage depends on every other stage having completed, should CheckingClusters depend on Network, Managed Identity, AKS, etc... ?
> 
> I thought it would be each stage depends on the one before it having succeeded?

thought the checking cluster would be the depend on all of them as it is pre-stage needed; however logically it make sense to only have AKS on it 



At a minimum, CheckingClusters needs to depend on BootstrapCluster, PipelineTests, and AKS stages. This is because BootstrapCluster and PipelineTests don't run for a PR and skip straight away, which could lead to the next environment starting prematurely. AKS always runs, so it determines when the next environment starts in the PR case.

Reply:
@NickAzureDevops did you change this in some stages now? I think i misread at first, and seems like Managed_Identity, AKS etc stages still need these dependencies as they are component.dependsOn (I missed dependsOn the first time round)

does this need adding back in some way as it seems removed now?

Reply:
Looks OK btw, but wanted to check as not seen this bit in a while